### PR TITLE
Remove query id in favor of query filename

### LINKED ISSUE #735: Consider either testing that lint file names match their `id` fields, or removing the field entirely.
Sometimes new lint contributions accidentally fail to set the `id` field of new lints correctly. #734 fixed one such issue.

We want a field like `id` available on the deserialized value for purposes of naming the lint and reporting failures.

We should either have a test that makes sure each lint file name matches its `id` field, or remove the field from the file and somehow otherwise populate it during deserialization.

### PR DESCRIPTION
**Pull request by pksunkara**
fixes #735,

### CODE REVIEW

File: src/query.rs [hunk 1]
```diff
@@ -462,10 +462,10 @@ macro_rules! add_lints {
             )*
         }
 
-        fn get_query_text_contents() -> Vec<&'static str> {
+        fn get_queries() -> Vec<&'static str, &'static str)> {
```
Comment (obi1kenobi) [inline src/query.rs:465]:
```suggestion
        fn get_queries() -> Vec<(&'static str, &'static str)> {
```

File: src/lints/union_field_missing.ron [hunk 2]
```diff
@@ -44,7 +43,7 @@ SemverQuery(
                         field @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
                             visibility_limit @filter(op: "=", value: ["$public"])
                             name @filter(op: "=", value: ["%field_name"])
-                            
+
```
Comment (obi1kenobi) [inline src/lints/union_field_missing.ron:46]:
```suggestion
```
let's just delete that line altogether, it isn't necessary

**Review by obi1kenobi — APPROVED** [review]

**Comment by obi1kenobi** [comment]
Thanks! Fixed a minor typo to help tests pass, and now queued for merge.

**Comment by pksunkara** [comment]
Trying to fix the test failures made me realize that this is the wrong approach because we are utilising `SemverQuery.id` in many places and passing the id around in all those places makes the code worse.

I would close this PR in favor of adding a test as described in #735 if you agree.

**Comment by obi1kenobi** [comment]
Sounds good!

**Pull request closed without merging**